File: Taking-Maximum-Energy-From-the-Mystic-Dungeon/Taking-Maximum-Energy-From-the-Mystic-Dungeon.py
energy = [5,2,-10,-5,1]
k = 3

def maximumEnergy(energy,k):
    for i in range(len(energy)-1-k,-1,-1):
        energy[i] = energy[i] + energy[i+k]            
    return max(energy)

    # maxs = energy[len(energy)-1]
    # for j in range(len(energy)-1,len(energy)-1-k,-1):
    #     maxs = max(maxs,energy[j])
    
    # for i in range(len(energy)-1-k,-1,-1):
    #     energy[i] = energy[i] + energy[i+k]
    #     maxs = max(maxs,energy[i])        
    # return maxs
print(maximumEnergy(energy,k))

File: Taking-Maximum-Energy-From-the-Mystic-Dungeon/Taking-Maximum-Energy-From-the-Mystic-Dungeon.js
let energy = [5,2,-10,-5,1], k = 3

var maximumEnergy = function(energy, k) {
    let maxs = energy[energy.length-1]
    for(let j = energy.length-1; j>energy.length-1-k;j--){
        maxs = Math.max(energy[j],maxs)
    }

    for( let i = energy.length-1-k ; i>=0;i--){
        energy[i]=energy[i]+energy[i+k]
        maxs = Math.max(energy[i],maxs)
    }
    return maxs

    /*
    Method 1
    for( let i = energy.length-1-k ; i>=0;i--){
        energy[i]=energy[i]+energy[i+k]
    }
    return Math.max(...energy)
    */

};

console.log(maximumEnergy(energy,k))